src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -291,12 +294,26 @@ function ScatterplotViz(props: VisualizationProps) {
       xAxisVariable,
       'scatterplot'
     );
-    return independentAxisRangeMargin(
-      defaultIndependentRange,
-      xAxisVariable?.type
-    );
+    return axisRangeMargin(defaultIndependentRange, xAxisVariable?.type);
   }, [xAxisVariable]);
 
+  // find deependent axis range and its margin
+  const defaultDependentRangeMargin = useMemo(() => {
+    //K set yMinMaxRange using yMin/yMax obtained from processInputData()
+    const yMinMaxRange =
+      data.value != null
+        ? { min: data.value.yMin, max: data.value?.yMax }
+        : undefined;
+
+    const defaultDependentRange = defaultDependentAxisRange(
```

Comment:
Do we have time to refactor `defaultDependentAxisRange` and `defaultIndependentAxisRange` into one function? It seems they do similar things.  The former has an extra range input (the [yMin, yMax] range from the actual data) - but the logic is the same: "given a bunch of ranges, calculate the union of them".  Also, instead of passing the plot type, we could pass a boolean directive saying whether or not zero should be including in the range-min calculations.

@dmfalke - should we add this to Technical Debt or do it now?  I think probably the former?

Reply:
@bobular Yes you are correct. I also thought of a combined function like you said, but it is a bit complicated to check all possible cases at once so I just separately made it at different function for faster implementation. In addition, perhaps a prop like isZeroMin or something like that may replace plotName prop as you said. 

I think we may refactor these two items later.

Reply:
Let's add it to Technical Debt. Thank!

Reply:
@dmfalke Thank you for your suggestion! I will merge this PR and make a technical debt one as well